scripts/modules_auto_pr/modules_auto_pr.sh

```diff
@@ -0,0 +1,98 @@
+#                                            VARIABLES                                             #
+####################################################################################################
+
+TIMESTAMP=$(date +"%Y-%m-%d_%H-%M")
+HUMAN_TIMESTAMP=$(date "+%-d %B %Y, %H:%M")
+REPO_URL="git@github.com:hpcugent/vsc_user_docs" # The repository for which the script will generate the PR
+BASE_BRANCH="main"
+BRANCH_NAME="auto_update_modules_$TIMESTAMP"
+REPO_NAME="vsc_user_docs" # name of the cloned repo. script available_modules.py requires this to be the name. Do not change.
+REPO_PATH="/tmp/$USER/modules_auto_pr_script_$TIMESTAMP/$REPO_NAME"
+COMMIT_MESSAGE="Update Modules [$HUMAN_TIMESTAMP]"
+PR_TITLE="Auto Update Modules [$HUMAN_TIMESTAMP]"
+
+make_pr_body() {
+  local n_added_modules="$1"
+  local n_removed_modules="$2"
+
+  echo "This is an automated pull request updating the markdown files of all available modules."
+  echo ""
+  echo "Changes:"
+  echo "- Updated the markdown files of all previously available modules"
+  echo "- Added $n_added_modules new modules"
+  echo "- Removed $n_removed_modules modules"
+}
+
+####################################################################################################
+#                                             SCRIPT                                               #
+####################################################################################################
+
+echo_info()  { echo -e "\e[32m$0: [INFO] $1\e[0m"; }
+echo_error() { echo -e "\e[31m$0: [ERROR] $1\e[0m"; }
+
+main() {
+  local github_pat_file="$1"
+  local fork_user="$2" # The user that owns the fork from which the script will create the PR. e.g. lbarraga
+  local fork_url="git@github.com:$fork_user/vsc_user_docs" # The fork from which the script will create the PR
+
+  # Check if the `gh` command is available
+  if ! command -v gh &> /dev/null; then
+      echo_error "The GitHub CLI (gh) is not installed."
+      echo_error "Please install it by following the instructions at https://cli.github.com."
+      exit 1
+  fi
+
+  # Check if PAT file is present and fork user is provided in one if statement
+  # -s checks whether file exists and has non-zero size (is not empty)
+  if [ ! -s "$github_pat_file" ] || [ -z "$fork_user" ]; then
+      echo_error "Please provide a valid GitHub Personal Access Token (PAT) file and fork user."
+      echo_error "Usage: $0 <github_pat_file> <fork_user>"
+      exit 1
+  fi
+
+  gh auth login --with-token < "$github_pat_file"
+
+  # Clone the repository and create a new branch
+  git clone $REPO_URL "$REPO_PATH"
+  cd "$REPO_PATH"
+  git checkout -b "$BRANCH_NAME"
+
+  # run available_software.py
+  python -m venv venv
+  source venv/bin/activate
+  pip install -r scripts/available_software/requirements.txt
+  python scripts/available_software/available_software.py
+
+  # Add and commit the generated files
+  git add .
```

Comment:
@lbarraga To be fixed in follow-up PR that tries to prevent silly timestamp-bump-only PRs: make this a bit more strict by only staging the changes in the subdirectory where the `available_software.py` script generates data, to try and avoid random untracked stuff from being shoved into the PR being created:

```suggestion
  # add and commit (only) the files generated by the available_software.py script
  git add mkdocs/docs/HPC/only/gent/available_software/
```

In the account that we'll use to run this script, I'll add a `cron.sh` and and extra `README` with some info in `scripts/[modules_auto_pr/` for example, but that should never be added into a PR.

Another example is the file that holds the GitHub token, we really do not want to accidentally include that into a PR 🙈 

That shouldn't happen, since a fresh clone is being made in a temporary directory, but we better try and avoid mistakes if more changes are being made to this script later (I can imagine us going "why do we clone from scratch if we already have a checkout of this repo, we can just copy+update that instead" in the future, for example).